packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -306,19 +316,35 @@ export class TeamsBotV2Impl {
   private resolveScenarios(
     actRoles: PluginActRoles[],
     triggers: BotTrigger[],
-    hostType: HostType
+    hostType: HostType,
+    isM365: boolean | undefined
   ): string[] {
     const scenarios: string[] = [];
-    if (hostType === HostType.Function) {
-      if (actRoles.includes(PluginActRoles.Notification)) {
-        scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
-        triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+    actRoles.map((actRole) => {
+      switch (actRole) {
+        case PluginActRoles.CommandAndResponse:
+          scenarios.push(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
+          break;
+        case PluginActRoles.Notification:
+          if (hostType === HostTypes.AZURE_FUNCTIONS) {
+            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
+            triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+          }
+          if (hostType === HostTypes.APP_SERVICE) {
+            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_RESTIFY_SCENARIO_NAME);
+          }
+          break;
+        case PluginActRoles.Bot:
+        case PluginActRoles.MessageExtension:
+          if (isM365) {
```

Comment:
It is weird to have this isM365 logic here, let's sync offline and find a better to organize them

Reply:
According to our discussion, I added the logic at the start of the `decideTemplateScenarios` function. Will check if it can be improved later.